tapfreighter/parcel.go

```diff
@@ -348,150 +348,201 @@ func ConvertToTransfer(currentHeight uint32, vPkt *tappsbt.VPacket,
 	for idx := range vPkt.Inputs {
 		vIn := vPkt.Inputs[idx]
 
-		// We don't know the actual outpoint the input is spending, so
-		// we need to look it up by the pkScript in the anchor TX.
-		var anchorOutPoint *wire.OutPoint
-		for inIdx := range anchorTx.FundedPsbt.Pkt.Inputs {
-			pIn := anchorTx.FundedPsbt.Pkt.Inputs[inIdx]
-			if pIn.WitnessUtxo == nil {
-				return nil, fmt.Errorf("anchor input %d has "+
-					"no witness utxo", idx)
-			}
-			utxo := pIn.WitnessUtxo
-			if bytes.Equal(utxo.PkScript, vIn.Anchor.PkScript) {
-				txIn := anchorTx.FinalTx.TxIn[inIdx]
-				anchorOutPoint = &txIn.PreviousOutPoint
-				break
-			}
-		}
-		if anchorOutPoint == nil {
-			return nil, fmt.Errorf("unable to find anchor "+
-				"outpoint for input %d", idx)
+		tIn, err := transferInput(anchorTx.FundedPsbt.Pkt, vIn)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert input %d: %w",
+				idx, err)
 		}
 
-		parcel.Inputs[idx] = TransferInput{
-			PrevID: asset.PrevID{
-				OutPoint: *anchorOutPoint,
-				ID:       vIn.Asset().ID(),
-				ScriptKey: asset.ToSerialized(
-					vIn.Asset().ScriptKey.PubKey,
-				),
-			},
-			Amount: vIn.Asset().Amount,
-		}
+		parcel.Inputs = append(parcel.Inputs, *tIn)
 	}
 
-	outputCommitments := anchorTx.OutputCommitments
 	for idx := range vPkt.Outputs {
-		vOut := vPkt.Outputs[idx]
-
-		// Convert any proof courier address associated with this output
-		// to bytes for db storage.
-		var proofCourierAddrBytes []byte
-		if vOut.ProofDeliveryAddress != nil {
-			proofCourierAddrBytes = []byte(
-				vOut.ProofDeliveryAddress.String(),
-			)
+		tOut, err := transferOutput(
+			vPkt, idx, anchorTx, passiveAssets, isLocalKey,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert "+
+				"output %d: %w", idx, err)
 		}
 
-		anchorInternalKey := keychain.KeyDescriptor{
-			PubKey: vOut.AnchorOutputInternalKey,
+		parcel.Outputs = append(parcel.Outputs, *tOut)
+	}
+
+	return parcel, nil
+}
+
+// transferInput creates a TransferInput from a virtual input and the anchor
+// packet.
+func transferInput(anchorPacket *psbt.Packet,
+	vIn *tappsbt.VInput) (*TransferInput, error) {
+
+	// We don't know the actual outpoint the input is spending, so we need
```

Comment:
Isn't this just in the prev ID section of the asset included in the `vIn`?

Reply:
Or I guess it just might not be populated at this point?

Reply:
Hmm, you're right. This should just be the prevID and it should be set at this point. Not sure why it was implemented this way initially.